sphinx_doc/tutorials/docs/writing_new_costmap2d_plugin.rst

```diff
@@ -0,0 +1,195 @@
+.. _writing_new_costmap2d_plugin:
+
+Writing a new Costmap2D plugin
+******************************
+
+- `Overview`_
+- `Requirements`_
+- `Tutorial Steps`_
+
```

Comment:
A short (5-10s) gif here would be helpful. That way at the start of the tutorial, they see what they're going to learn about and the final result.